runtime/interpreter/big.go

```diff
@@ -140,3 +141,46 @@ func BigEndianBytesToSignedBigInt(b []byte) *big.Int {
 func BigEndianBytesToUnsignedBigInt(b []byte) *big.Int {
 	return new(big.Int).SetBytes(b)
 }
+
+func BigIntSqrt(interpreter *Interpreter, value *big.Int, locationRange LocationRange) UFix64Value {
+	if value.Sign() < 0 {
+		panic(UnderflowError{
+			LocationRange: locationRange,
+		})
+	}
+
+	if value.Cmp(sema.MaxSquareIntegerBig) == 1 {
```

Comment:
When calling this check in the case of `Fix64Value` and `UFix64Value`, I think we would exclude some values because `value` is larger than the max, but in reality it represents a number less than `max` (because `value` is multiplied by the factor). Actually, I think all `Fix64` and `UFix64` values are less than the `max`, so the result (square root) can always fit.

If I didn't miss anything and my comment makes sense, it may be better to have a specific function `BigIntSqrtFloat` to process the case of `UFix64` and `Fix64` skipping the overflow (also to add the optimization from my previous comment)